app_dart/test/request_handling/static_file_handler_test.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:cocoon_service/cocoon_service.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import 'package:cocoon_service/src/request_handling/body.dart';
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+
+import '../src/datastore/fake_cocoon_config.dart';
+import '../src/request_handling/request_handler_tester.dart';
+
+void main() {
+  group('StaticFileHandler', () {
+    RequestHandlerTester tester;
+    StaticFileHandler<Body> staticFileHandler;
+
+    final FakeConfig config = FakeConfig();
+
+    setUp(() {
+      tester = RequestHandlerTester();
+    });
+
+    test('returns 404 response when file does not exist', () async {
```

Comment:
@tvolkert should I add tests for mocking the filesystem? I would like to be able to verify a file response is correctly sent, but I am having a hard time finding resources on that topic in Dart 2.

Reply:
It might be a bit complicated for this, but there is always IOOverrides, see

https://github.com/flutter/flutter/blob/c8b3c9b7270bf6d768899fd78bd5b40466514334/packages/flutter_tools/test/src/io.dart#L5

Reply:
See comment above about injecting a `FileSystem` into the handler.